addons/nuxeo-adobe-connector/nuxeo-adobe-connector-core/src/test/java/org/nuxeo/adobe/cc/TestPageProviders.java

```diff
@@ -0,0 +1,179 @@
+        DocumentModel collection = cm.createCollection(session, "my-collec", "dummy", testWorkspacePath);
+        cm.addToCollection(collection, doc, session);
+
+        AtomicReference<String> collectionId = new AtomicReference<>();
+        testPageProvider("adobe-connector-other_primary", (entries) -> {
+            assertThat(entries).hasSize(1);
+            collectionId.set(entries.get(0).get("uid").asText());
+        });
+
+        DocumentModel finalDoc = doc;
+        testPageProvider("adobe-connector-other_secondary", (entries) -> {
+            assertThat(entries).hasSize(1);
+            assertThat(entries.get(0).get("uid").asText()).isEqualTo(finalDoc.getId());
+        }, collectionId.get());
+    }
+
+    @Test
+    public void testSearch() throws Exception {
+        DocumentModel doc = session.createDocumentModel(testWorkspacePath, "foo", PICTURE_TYPE_NAME);
+        doc.setPropertyValue("dc:description", "hello");
+        session.createDocument(doc);
+
+        doc = session.createDocumentModel(testWorkspacePath, "bar", PICTURE_TYPE_NAME);
+        doc.setPropertyValue("dc:description", "world");
+        doc = session.createDocument(doc);
+
+        Map<String, String> params = new HashMap<>();
+        params.put("system_fulltext", "world");
+        params.put("system_parentId", testWorkspaceId);
+
+        DocumentModel finalDoc = doc;
+        testPageProvider("adobe-connector-search", (entries) -> {
+            assertThat(entries).hasSize(1);
+            assertThat(entries.get(0).get("uid").asText()).isEqualTo(finalDoc.getId());
+        }, params);
+    }
+
+    protected void testPageProvider(String ppName, Consumer<List<JsonNode>> consumer, Map<String, String> qarams,
+            String... parameters) throws Exception {
+        awaitCompletion();
+
+        MultivaluedMap<String, String> queryParams = new MultivaluedMapImpl();
+        queryParams.put(ORDERED_PARAMS, Arrays.asList(parameters));
+        qarams.forEach(queryParams::putSingle);
+
+        try (CloseableClientResponse res = getResponse(RequestType.GET, "search/pp/" + ppName + "/execute",
+                queryParams)) {
+            assertEquals(Response.Status.OK.getStatusCode(), res.getStatus());
+            List<JsonNode> logEntries = getLogEntries(mapper.readTree(res.getEntityInputStream()));
+
+            consumer.accept(logEntries);
+        }
+    }
+
+    protected void testPageProvider(String ppName, Consumer<List<JsonNode>> consumer, String... parameters)
+            throws Exception {
+        testPageProvider(ppName, consumer, Collections.emptyMap(), parameters);
+    }
+
+    protected void awaitCompletion() throws InterruptedException, TimeoutException, ExecutionException {
+        // Save session, and await for all async completion (ES indexing...)
+        session.save();
+        tf.nextTransaction();
+
+        workManager.awaitCompletion(20, TimeUnit.SECONDS);
```

Comment:
Not needed, it is done in `nextTransaction`. Currently we have 4 waiters we're waiting in `nextTransaction`:
- work manager
- baf
- ES audit in drive (with ESAuditFeature)
- audit (wait for bulker or stream, index refreshing is not in this scope)

Reply:
Cool, I didn't know that! I extracted a dedicated waiter for `RepositoryElasticSearchFeature`!

Reply:
Really cool :v: thanks!